### PR TITLE
Modify param data for SLES artifacts based on native build support before returning package info

### PR DESCRIPTION
**Pull request by chef-delivery**
The primary change here is to properly map platform and platform versions for SLES requests based on a products version. This opened up a can worms while trying to update tests. The test data was so out of date and corrupt. It was making troubleshooting and comparing results more difficult than it needed to be.

So... I updated ALL the test data to represent real data with the new URL paths. It made adding new tests much easier, although a bit time consuming. 

The new SLES tests added 252 new specs. The other smaller changes either represent newer data or fixed issues where a single value (sha1) for example was copied and pasted throughout the test data. There were other examples of forced data manipulation that I fixed.

Next time there are big changes it should be easier to wrap our heads around failing tests.

I gave `#get_package_info ` in `app.rb` a face lift. It's growing in complexity and there was a need for reuse of code. Not perfect, but improved.
 
This Pull Request was opened by Chef Automate user Patrick Wright

Signed-off-by: Patrick Wright <patrick@chef.io>

Ready to merge? View this change in Chef Automate and click the Approve button: https://automate.chef.co/e/chef/#/organizations/CIA/projects/omnitruck/changes/ed7e3b5f-5bfb-4c41-8099-a1d33b7ffda5

@chef/engineering-services

### CODE REVIEW

**Comment by wrightp** [comment]
Weird. I squashed the commits locally, but they still showed up in the automate review.

**Comment by wrightp** [comment]
I'll need help debugging the Automate build errors

**Comment by schisamo** [comment]
@wrightp Is the plan to eventually push this logic up into mixlib-install?

**Comment by wrightp** [comment]
@schisamo Yes. In the case I needed to figure out how to solve the issue before moving the logic. Moving the platform remapping logic will be more of a re-write. I'm am writing up the requirements and explanations as to why.

**Comment by wrightp** [comment]
Current build failure: https://gist.github.com/wrightp/14853749576d7282917e37d5d3386f5e

File: .delivery/build-cookbook/Berksfile [hunk 1]
```diff
@@ -9,3 +9,6 @@ cookbook 'oc-users', git: 'git@github.com:opscode-cookbooks/oc-users.git'
 cookbook 'route53', git: 'https://github.com/cwebberOps/route53.git', branch: 'pdb/nononokogiri'
 cookbook 'brightbox-ruby', github: 'Padgett/brightbox-ruby-cookbook', branch: 'boolean-generate-binstubs'
 cookbook 'habitat-build', git: 'https://github.com/chef-cookbooks/habitat-build'
+cookbook 'expeditor',
+         git: 'git@github.com:chef-cookbooks/expeditor.git',
+         ref: 'v0.1.15' # temporary lock until 1.5 release
```
Comment (tduffield) [inline .delivery/build-cookbook/Berksfile]: You don't need this line. This was just so I didn't break Automate.

File: .expeditor/config.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Slack channel in Chef Software slack to send notifications about build failures, etc
+slack:
+  notify_channel: eng-services-rfr
```
Comment (tduffield) [inline .expeditor/config.yml]: We've been doing eng-services-alerts for this channel.

**Pull request closed without merging**